src/java/main/org/apache/zookeeper/server/persistence/FileTxnSnapLog.java

```diff
@@ -167,6 +175,16 @@ public long restore(DataTree dt, Map<Long, Integer> sessions,
             PlayBackListener listener) throws IOException {
         long deserializeResult = snapLog.deserialize(dt, sessions);
         FileTxnLog txnLog = new FileTxnLog(dataDir);
+        boolean suspectEmptyDB;
+        File initFile = new File(dataDir.getParent(), "initialize");
+        if (initFile.exists()) {
+            if (!initFile.delete()) {
+                throw new IOException("Unable to delete initialization file " + initFile.toString());
+            }
+            suspectEmptyDB = false;
+        } else {
+            suspectEmptyDB = !autoCreateDB;
```

Comment:
IMO, it would be nice to put a `debug` (warn?) log message here. Something along the lines of "Initialize file doesn't found! Using autoCreateDB attribute."